packages/fiori/src/Bar.js

```diff
@@ -133,6 +133,34 @@ class Bar extends UI5Element {
 			"label": this.design,
 		};
 	}
+
+	resizeHandler(entry) {
+		const bar = entry[0].target;
+		let changeFlex = false;
+		const barWidth = bar.offsetWidth;
+
+		for (let i = 0; i < bar.childElementCount; i++) {
+			if (barWidth / 3 < bar.children[i].offsetWidth) {
+				changeFlex = true;
+			}
+		}
+
+		if (changeFlex) {
+			for (let i = 0; i < bar.childElementCount; i++) {
+				bar.children[i].classList.add("ui5-bar-content-container-overflow");
+			}
+		} else {
+			for (let i = 0; i < bar.childElementCount; i++) {
+				bar.children[i].classList.remove("ui5-bar-content-container-overflow");
+			}
+		}
+	}
+
+	onAfterRendering() {
+		if (!this.sizeObserver) {
```

Comment:
We have a `ResizeHandler.js` class that wraps the native ResizeObserver and provides a polyfill for IE. Please use this class. Also, the best practice is to use onEnterDOM and onExitDOM to register/deregister it. onAfterRendering will run even if the component is not in the dom, just created with document.createElement.

See some other components for the usage of ResizeHandler overall.